src/Microsoft.DotNet.Interactive.PostgreSQL/PostgreSqlKernel.cs

```diff
@@ -0,0 +1,328 @@
+            }
+        }
+
+        private IEnumerable<IEnumerable<IEnumerable<(string name, object value)>>> GetEnumerableTables(ColumnInfo[] columnInfos, CellValue[][] rows)
+        {
+            var displayTable = new List<(string, object)[]>();
+            var columnNames = columnInfos.Select(info => info.ColumnName).ToArray();
+
+            SqlKernelUtils.AliasDuplicateColumnNames(columnNames);
+
+            foreach (CellValue[] row in rows)
+            {
+                var displayRow = new (string, object)[row.Length];
+
+                for (var colIndex = 0; colIndex < row.Length; colIndex++)
+                {
+                    object convertedValue = default;
+
+                    try
+                    {
+                        var columnInfo = columnInfos[colIndex];
+
+                        var expectedType = Type.GetType(columnInfo.DataType);
+
+                        if (TypeDescriptor.GetConverter(expectedType) is { } typeConverter)
+                        {
+                            if (typeConverter.CanConvertFrom(typeof(string)))
+                            {
+                                // TODO:fix handling target boolean type when the column is bit type with numeric value
+                                if ((expectedType == typeof(bool) || expectedType == typeof(bool?)) &&
+
+                                    decimal.TryParse(row[colIndex].DisplayValue, out var numericValue))
+                                {
+                                    convertedValue = numericValue != 0;
+                                }
+                                else
+                                {
+                                    convertedValue =
+                                        typeConverter.ConvertFromInvariantString(row[colIndex].DisplayValue);
+                                }
+                            }
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        convertedValue = row[colIndex].DisplayValue;
+                    }
+
+                    displayRow[colIndex] = (columnNames[colIndex], convertedValue);
+                }
+
+                displayTable.Add(displayRow);
+            }
+
+            yield return displayTable;
+        }
+
+        public async Task HandleAsync(RequestCompletions command, KernelInvocationContext context)
+        {
+            if (!_intellisenseReady)
+            {
+                return;
+            }
+
+            var completionItems = await _serviceClient.ProvideCompletionItemsAsync(_tempFileUri, command);
+            context.Publish(new CompletionsProduced(completionItems, command));
+        }
+
+        protected override ChooseKernelDirective CreateChooseKernelDirective() =>
+            new ChoosePostgreSqlKernelDirective(this);
+
+        private class ChoosePostgreSqlKernelDirective : ChooseKernelDirective
```

Comment:
The `ChooseKernelDirective` and other usage of System.CommandLine has been removed from the `main` branch as of a couple months ago. You can read more about this here: https://github.com/dotnet/interactive/issues/3567.

The new implementation is available in VS Code Insiders.